R/shinyapp.R

```diff
@@ -345,6 +345,17 @@ loadSupport <- function(appDir=NULL, renv=new.env(parent=globalenv()), globalren
     appDir <- findEnclosingApp(".")
   }
 
+  descFile <- file.path.ci(appDir, "DESCRIPTION")
+  if (file.exists(descFile) &&
+      identical(as.character(read.dcf(descFile, fields = "Type")), "Package"))
```

Comment:
I'm pretty sure `Type` isn't required and defaults to `Package`, so we should also warn if it's missing?

Reply:
Hm, the problem is that Shiny apps can have a DESCRIPTION file with no `Type` field.

How about we also check for a NAMESPACE file?

Reply:
> Hm, the problem is that Shiny apps can have a DESCRIPTION file with no Type field.

Packages, even on CRAN, can as well (plotly doesn't, for example)

> How about we also check for a NAMESPACE file? I think almost all packages have NAMESPACE files these days.

Sounds good to me

Reply:
>> Hm, the problem is that Shiny apps can have a DESCRIPTION file with no Type field.

> Packages, even on CRAN packages, can as well (plotly doesn't, for example)

Right, my point was that if we warned if there was a missing Type field, many legitimate Shiny apps would get a warning.